### PR TITLE
Tutorial for stereoisomerism and QSAR

### PR DESCRIPTION
**Pull request by zhu0619**
## Changelogs
This PR adds a tutorial that explains the significance of stereoisomers exhibiting activity cliffs in a dataset and their impact on QSAR modeling.

This notebook covers exploration of different types of stereoisomerization, how stereochemical information can be captured through molecular representations. We also examine the impact of stereoisomers that exhibit significant activity cliffs (ACs) in QSAR modeling.

### CODE REVIEW

**Review by cwognum — APPROVED** [review]
This is awesome, @zhu0619 ! Thanks for pulling this together! I actually learned a lot reading through this myself.

I'll leave some comment and questions I had while reading down below: 
- Is the first cell actually tagged? If not, see [here](https://jupyterbook.org/en/stable/content/metadata.html#adding-tags-using-notebook-interfaces). You should use the [one of these tags](https://github.com/polaris-hub/polaris/blob/main/mkdocs.yml#L124-L127).
```
# Note: Cell is tagged to not show up in the mkdocs build
```
- What is the difference in spatial arrangement for enantiomers ? Could we visualize this in 3D as well to explain?
- In addition to `dm.same_mol`, could we add a little bit more context on the value of different equality tests? E.g. will InCHI keys detect stereochemistry? Are there cases where you don't need to (or even shouldn't) account for stereochemistry?
-  Similar question for E/Z isomers. I don't see these in the (very helpful! :fire: ) figure at the top either? 
- I would briefly introduce `molfeat`. 
- I would briefly introduce the task you're modeling and explain why 3D features are important for this task.
- I would hide the long output of cell `[33]`. You can keep the logs, but hide it in the docs by tagging the cell with `remove_output`.
- Could you provide more info on the importance of predicting ACs to contextualize the results, e.g. by giving one example in which you care about accurately predicting ACs and one example where you care more about overall performance?

**Comment by zhu0619** [comment]
@cwognum Haven't thought you will review so quickly. 
Thanks for the feedback. Good ideas. 
I also asked Ivan to review the notebook.

**Comment by Fransu86** [comment]
I'm working with Lu to correct some aspects of the document and add one additional edge case (mixed defined, undefined stereochemistry), as well as remake the figure from Wikipedia since it is horrible and actually has a couple of mistakes LOL

**Comment by zhu0619** [comment]
@Fransu86 It's ready for your review. :)

File: docs/tutorials/utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+from rdkit.Chem import AllChem
+
+import datamol as dm
+import pandas as pd
+
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import classification_report
+
+
+def add_3d_coord(mol: dm.Mol):
```
[the remark this answers is not in the document]
Comment (zhu0619) [reply docs/tutorials/utils.py:13]: yeah, but it took much longer to process. 
But I added this function as optional.

**Pull request merged**